hudi-flink/src/main/java/org/apache/hudi/sink/StreamWriteOperatorCoordinator.java

```diff
@@ -291,17 +296,11 @@ private void initHiveSync() {
 
   private void syncHiveIfEnabled() {
     if (tableState.syncHive) {
-      this.hiveSyncExecutor.execute(this::syncHive, "sync hive metadata for instant %s", this.instant);
+      LOG.info(String.format("sync hive metadata for instant %s", this.instant));
+      hiveSyncContext.hiveSyncTool().syncHoodieTable();
     }
```

Comment:
Do not modify this method and add a new method `syncHiveSynchronously` which is used for endInput event specially.